src/site/shared/api/Circuits.ts

```diff
@@ -2,23 +2,23 @@ import {CircuitMetadata, CircuitMetadataDef} from "core/models/CircuitMetadata";
 import {Request} from "shared/utils/Request";
 import {AuthState} from "./auth/AuthState";
 
+//
+// export async function CreateUserCircuit(auth: AuthState, data: string): Promise<CircuitMetadata | undefined> {
+//     return new CircuitMetadata(JSON.parse(await Request({
+//         method: "POST",
+//         url: "api/circuits",
+//         headers: {
+//             "authType": auth.getType(),
+//             "authId": auth.getId()
+//         },
+//         data
+//     })) as CircuitMetadataDef);
+// }
 
-export async function CreateUserCircuit(auth: AuthState, data: string): Promise<CircuitMetadata | undefined> {
+export async function UpdateUserCircuit(auth: AuthState, data: string, circuitId: string): Promise<CircuitMetadata | undefined> {
```

Comment:
The `circuitId` here feels like the right place to use a nullable string rather than rely on empty string as a special value.  Is that idiomatic @LeonMontealegre.  It might have other implications though.

Reply:
Agreed, yeah use `circuitId?: string` and pass undefined when it's not defined